### DISCUSSION TITLE
ESP32 multi channel ADC

### DISCUSSION THREAD

**Discussion by antoniomolram** [arendst/Tasmota#12775] [Q&A] [answered]
Hi guys,

first of all, thank you for this great piece of code. 

I was just asking myself if a multiple channel ADC make sense for you. As per Documentation of the ESP32 there are 2 ADC channels availables with multiple pins. This pins can be even measured in a multiplexed way.

https://docs.espressif.com/projects/esp-idf/en/latest/esp32/api-reference/peripherals/adc.html

What do you think?

**Comment by sfromis** [comment, marked as answer]
Tasmota does already support multiple (up to 8) ESP32 ADC channels, on gpio 32 and up. Those pins where the code to the left in "Configure Module" includes an `A`
It looks like ADC1 is being used for this, not ADC2 (due to ADC2 interfering with Wifi).

**Comment by antoniomolram** [comment]
hi, exactly. I was able to use it after you mention this, but there are functionalities that I cannot use then. For example, on the ADC range mapping there is no way to change the ADC that is going to be mapped, if I am wrong, how? 

https://tasmota.github.io/docs/Commands/ 
adcparam

**Reply by sfromis** [reply]
While default for [AdcParam](https://tasmota.github.io/docs/Commands/#adcparam) is the first channel, you can add an index to the command name, meaning that, say, `AdcParam3 ....` will address ADC channel 3. Obviously, this needs at least 3 channels configured, and as the index is in sequence, it does not make sense to try to skip numbers in the pin assignments.

**Reply by antoniomolram** [reply]
Thank you for the info, this was something that wasn't clear for me. I am using it now! Thanks!
